src/utils/clusterUtils.js

```diff
@@ -1,9 +1,28 @@
 import cmp from 'semver-compare';
-import { Providers } from 'shared/constants';
+import { Constants, Providers } from 'shared/constants';
 
 // Here we can store functions that don't return markup/UI and are used in more
 // than one component.
 
+// Determine whether a cluster at a certain version can be upgraded to a target version.
+export function canClusterUpgrade(currentVersion, targetVersion, provider) {
+  // Cluster must have a release_version.
+  if (currentVersion === '') return false;
+  if (Boolean(currentVersion) !== true) return false;
```

Comment:
Why do we need both of these checks here?

Shouldn't `if (!currentVersion) return false` be enough?

Reply:
I think theres some weird API corner case we once had to guard against, where we'd get a empty string back. It might not be a issue anymore though, but I am reluctant to change that right now 😅

Reply:
`if (!currentVersion) return false;` guards against an empty string, `null`, `undefined`, `0`, `NaN` and `false` so we should be safe 👍

Reply:
also I do prefer `(!currentVersion)` over `Boolean(currentVersion) !== true)`

But you started it :P ;)

https://github.com/giantswarm/happa/blame/228a39be86d3e2b3e5470570999e1079c9041884/src/components/Cluster/ClusterDetail/ClusterDetailView.js#L151

Reply:
Don't throw me under the bus! I just converted `!!this.props.targetRelease` to `Boolean(this.props.targetRelease)` because it was upsetting the linter 😬